src/gui/socketapi/socketuploadjob.cpp

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright (C) by Hannah von Reth <hannah.vonreth@owncloud.com>
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY
+ * or FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License
+ * for more details.
+ */
+
+#include "socketuploadjob.h"
+#include "socketapi_p.h"
+
+#include "accountmanager.h"
+#include "common/syncjournaldb.h"
+#include "syncengine.h"
+
+#include <QFileInfo>
+#include <QJsonArray>
+#include <QRegularExpression>
+
+using namespace OCC;
+
+SocketUploadJob::SocketUploadJob(OCC::SocketListener *listener, const QString &argument, QObject *parent)
+    : QObject(parent)
+    , _listener(listener)
+{
+    const auto args = QJsonDocument::fromJson(argument.toUtf8()).object();
+    _localPath = args[QLatin1String("localPath")].toString();
+    _remotePath = args[QLatin1String("remotePath")].toString();
+    _pattern = args[QLatin1String("pattern")].toString();
+    // TODO: use uuid
+    const auto accname = args[QLatin1String("account")][QLatin1String("name")].toString();
```

Comment:
Nitpick: QLatin1String -> QStringLiteral ?

Reply:
Json has an overload and as the string is not actually used, besides the lookup, QLatin1String should be slightly better 🙈

Reply:
https://forum.qt.io/topic/78540/qstringliteral-vs-qlatin1string/3

I think in the end it doesn't matter. This is unlikely to ever become a "hot path"

Reply:
https://code.woboq.org/qt5/qtbase/src/corelib/text/qstring.h.html#QLatin1Literal